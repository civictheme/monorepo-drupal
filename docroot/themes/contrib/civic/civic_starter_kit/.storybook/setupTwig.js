const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
