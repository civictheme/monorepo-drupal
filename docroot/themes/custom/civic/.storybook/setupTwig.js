const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
