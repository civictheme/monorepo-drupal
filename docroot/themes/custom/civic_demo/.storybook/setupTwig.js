const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
