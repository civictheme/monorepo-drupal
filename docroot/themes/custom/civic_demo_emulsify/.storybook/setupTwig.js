const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
