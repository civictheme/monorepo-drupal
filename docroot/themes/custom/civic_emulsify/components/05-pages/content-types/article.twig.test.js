import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import { setupTwig, namespaces } from '../../../.storybook/setupTwig';

setupTwig(Twig);

describe('article', () => {
  it('can render an article page', async () => {
    const { container } = await render(
      join(__dirname, 'article.twig'),
      {},
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });
});
