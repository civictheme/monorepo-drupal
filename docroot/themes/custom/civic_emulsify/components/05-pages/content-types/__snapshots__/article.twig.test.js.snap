// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article can render an article page 1`] = `
<div>
  


    

  <div
    class="layout-container"
  >
    
      


    <header
      class="header"
    >
      
  
      <div
        class="header__inner"
      >
        
    
        <div
          class="header__primary"
        >
          
      
          <div
            class="header__branding"
          >
            
                  


            <a
              class="logo-link"
              href="/"
            >
              
      


  

              <img
                alt="Logo"
                class="logo__image"
                src="logo.png"
              />
              
  
            </a>
            
              
          </div>
          
      
          <div
            class="header__menu"
          >
            
                  


            <nav>
              
  
              <a
                class="toggle-expand"
                href="#"
                id="toggle-expand"
              >
                
    
                <span
                  class="toggle-expand__open"
                >
                  
      






                  <svg
                    class="toggle-expand__icon"
                  >
                    
      
                    <use
                      xlink:href="icons.svg#menu"
                    />
                    

                  </svg>
                  
      
                  <span
                    class="toggle-expand__text"
                  >
                    Main Menu
                  </span>
                  
    
                </span>
                
    
                <span
                  class="toggle-expand__close"
                >
                  
      
                  <span
                    class="toggle-expand__text"
                  >
                    Close
                  </span>
                  
    
                </span>
                
  
              </a>
              
  
              <div
                class="main-nav"
                id="main-nav"
              >
                
    




    

              </div>
              

            </nav>
            
              
          </div>
          
    
        </div>
        
    
        <div
          class="header__secondary"
        >
          
              

          
        </div>
        
  
      </div>
      

    </header>
    
        
    <div
      class="main main--with-sidebar"
    >
      
    
      <a
        id="main-content"
        tabindex="-1"
      />
      
          
      <aside
        class="main-sidebar"
        role="complementary"
      >
        
            
        <div
          class="cta"
        >
          
  
          <h2>
            We have a podcast
          </h2>
          
  




          <button
            class="button"
          >
            
      Subscribe now!
  
          </button>
          

        </div>
        

        
      </aside>
      
        
      <main
        class="main-content main-content--with-sidebar"
        role="main"
      >
        
          


        <h2
          class="h2"
        >
          
      Page Intro
  
        </h2>
        
    


        <p
          class="paragraph"
        >
          
      At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum. Stet clita kasd gubergren, no sea takimata sanctus est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, At accusam aliquyam diam diam dolore dolores duo eirmod eos erat, et nonumy sed tempor et et invidunt justo labore Stet clita ea et gubergren, kasd magna no rebum. sanctus sea sed takimata ut vero voluptua. est Lorem ipsum dolor sit amet. Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat.
  
        </p>
        
    
  

  
  


        <h2
          class="h2"
        >
          
      Here's an awesome grid
  
        </h2>
        

        <div
          class="grid grid--card"
        >
          
              


          <div
            class="card grid__item"
          >
            
  
            


  

            <img
              alt="People"
              class="image"
              src="https://placeimg.com/480/300/people"
            />
            
        
  
            <div
              class="card__content"
            >
              
    
          
      


              <h2
                class="card__heading"
              >
                
      


                <a
                  class="card__heading-link"
                  href="#"
                >
                  
      Super Awesome Card
  
                </a>
                
  
              </h2>
              
      
        
          


              <h3
                class="card__subheading"
              >
                
      Person of Interest
  
              </h3>
              
        
          


              <p
                class="card__body"
              >
                
      Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Donec rutrum congue leo eget malesuada. Proin eget tortor risus. Sed porttitor lectus nibh. Nulla quis lorem ut libero malesuada feugiat. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Pellentesque in ipsum id orci porta dapibus. Nulla quis lorem ut libero malesuada feugiat. Nulla quis lorem ut libero malesuada feugiat.
  
              </p>
              
        
          


              <a
                class="card__link"
                href="#"
              >
                
      
  
              </a>
              
        
      
            </div>
            

          </div>
          
            


          <div
            class="card grid__item"
          >
            
  
            


  

            <img
              alt="Nature"
              class="image"
              src="https://placeimg.com/480/300/nature"
            />
            
        
  
            <div
              class="card__content"
            >
              
    
          
      


              <h2
                class="card__heading"
              >
                
      Card Number Two
  
              </h2>
              
      
        
          


              <h3
                class="card__subheading"
              >
                
      Some pretty sweet nature
  
              </h3>
              
        
          


              <p
                class="card__body"
              >
                
      Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Donec rutrum congue leo eget malesuada. Proin eget tortor risus. Sed porttitor lectus nibh. Nulla quis lorem ut libero malesuada feugiat. Vivamus magna justo.
  
              </p>
              
        
        
      
            </div>
            

          </div>
          
            


          <div
            class="card grid__item"
          >
            
  
            


  

            <img
              alt="Animals"
              class="image"
              src="https://placeimg.com/480/300/animals"
            />
            
        
  
            <div
              class="card__content"
            >
              
    
          
      


              <h2
                class="card__heading"
              >
                
      The Last Card Heading
  
              </h2>
              
      
        
          


              <h3
                class="card__subheading"
              >
                
      Awesome animals
  
              </h3>
              
        
          


              <p
                class="card__body"
              >
                
      Tempus convallis quis ac lectus. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Donec rutrum congue leo eget malesuada. Proin eget tortor risus. Sed porttitor lectus nibh. Nulla quis lorem ut libero malesuada feugiat.. Nulla quis lorem ut libero malesuada feugiat.
  
              </p>
              
        
        
      
            </div>
            

          </div>
          
      
        </div>
        
    
        <div
          class="cta"
        >
          
  
          <h2>
            Here's a cool thing!
          </h2>
          
  




          <button
            class="button"
          >
            
      Click me!
  
          </button>
          

        </div>
        
      
      </main>
      
  
    </div>
    
      

    <footer
      class="footer"
    >
      
  
      <div
        class="footer__inner"
      >
        
    
        <div
          class="footer__social"
        >
          
              


          <ul
            class="social-menu"
          >
            
        
          </ul>
          
          
        </div>
        
    
        <div
          class="footer__menu"
        >
          
              




    
          
        </div>
        
  
      </div>
      

    </footer>
    
  
  </div>
  

</div>
`;
